### ISSUE TITLE
[BUG] jQuery URLs throw error when server attempts to cache them

### ISSUE THREAD

**Issue by chaoticbear** [burningmantech/ranger-ims-server#1164]
Server logs show an error
```
[ims.application._external.ExternalApplication#critical] Download failed for https://code.jquery.com/jquery-3.1.0.min.js: 421 Misdirected Requestserver-1
[ims.application._external.ExternalApplication#error] Unable to open file /srv/ims/data/cache/jquery-3.1.0.min.js: [Errno 2] No such file or directory: '/srv/ims/data/cache/jquery-3.1.0.min.js'
```

The jQuery URLs in `src/ims/application/_external.py` line 701 & 705 are related. When switching to `http://` urls the caching appears to work. But that is not a good fix. This may require a python package update in order to get the latest cypher suites or something similar.

**Comment by wsanchez** [comment]
`421 Misdirected` looks like the cause…

`curl` can fetch `https://code.jquery.com/jquery-3.1.0.min.js` no problem… also Safari.

[This article](https://webidextrous.com/how-to-solve-421-misdirected-request-errors/) suggests it's a server config issue when HTTP/2 is in use.

Perhaps this is a transient issue… a work-around is to use `curl` or `wget` or whatever to fetch `https://code.jquery.com/jquery-3.1.0.min.js` and put it into your cache directory, in this case at `/srv/ims/data/cache/jquery-3.1.0.min.js`.

**Comment by toddgardner** [comment]
@wsanchez what would your feelings be at doing the fetch at docker image build time, when the rest of the third party deps are fetched -- code similar to the last PR, just with an added docker file change and removing the jquery from the repo

**Comment by chaoticbear** [comment]
@toddgardner @wsanchez I think a similar idea that is perhaps more standard than having Docker itself or Python do that work: 

 1. In the same repo create a small node project that imports jQuery & any other relevant JS libraries used in the legacy front-end.
 2. Automate the build of that project with a GH action or CI flow on every deployment.
 3. Remove the python caching logic and ensure the templates link the built JS from the project.
 4. Depend-a-bot should catch any security updates for those dependencies and create PRs for us.

**Comment by chaoticbear** [comment]
@toddgardner I'm going to work on a POC of my idea unless you're in process on something here.

**Comment by toddgardner** [comment]
I was just going to add a curl/wget of the existing URL to the docker-build but if you want to work on yours I'm not tied to it.

**Comment by chaoticbear** [comment]
Why don't you work on your version @toddgardner I don't need to take this from you :)

**Comment by wsanchez** [comment]
Unblocked by https://github.com/burningmantech/ranger-ims-server/pull/1176

**Comment by chaoticbear** [comment]
Closing as fixed with #1184  merged.
